posts/postRouter.js

```diff
@@ -1,27 +1,85 @@
 const express = require('express');
+const posts = require('./postDb');
 
 const router = express.Router();
 
 router.get('/', (req, res) => {
-  // do your magic!
+  posts.get()
+    .then(posts => {
+      res.status(200).json(posts)
+    })
+    .catch(err => {
+      res.status(500).json({
+        errorMessage: "Error fetching posts."
+      })
+    })
 });
 
-router.get('/:id', (req, res) => {
-  // do your magic!
+router.get('/:id', validatePostId(), (req, res) => {
+  posts.getById()
+    .then(post => {
+      res.status(200).json(post)
+    })
+    .catch(err => {
+      res.status(500).json({
+        errorMessage: "Error retrieving the post."
+      })
+    })
 });
 
-router.delete('/:id', (req, res) => {
-  // do your magic!
+router.delete('/:id', validatePostId(), (req, res) => {
+  const { post_id } = req.params.id;
```

Comment:
the syntax here isn't going to work.  It's saying you have `req.params.id.post_id`

Reply:
you need `const post_id = req.params.id`